src/util/heap/freelistpageresource.rs

```diff
@@ -311,29 +251,25 @@ impl<VM: VMBinding> FreeListPageResource<VM> {
     ) -> i32 {
         let mut rtn = freelist::FAILURE;
         let required_chunks = crate::policy::space::required_chunks(pages);
-        let region = self
-            .inner()
-            .common
-            .grow_discontiguous_space(space_descriptor, required_chunks);
+        let maybe_rmfl = sync.free_list.downcast_mut::<RawMemoryFreeList>();
+        let region =
+            self.common
+                .grow_discontiguous_space(space_descriptor, required_chunks, maybe_rmfl);
```

Comment:
We could use `Option<&mut dyn FreeList>` instead of `Option<&mut RawMemoryFreeList>` for `grow_discontiguous_space` and `allocate_contiguous_chunks`. In that case, the downcast happens in `VMMap`.

Reply:
I made the change as you suggested.  It makes sense because if the page resource uses free list, and Map64 can grow the space, they always need to grow the free list anyway.  It just requires `Map64` to choose a growable `FreeList` implementation.